### PR TITLE
Convert "Invalid operation while connection is closing" to retriable ServiceBusException.

### PR DESCRIPTION
**Pull request by DorothySun216**
Convert "Invalid operation while connection is closing" to retriable ServiceBusException so the SDK retry logic could kick in automatically. 

Reference PR: https://github.com/Azure/azure-sdk-for-net/pull/15984
Github Issues: https://github.com/Azure/azure-sdk-for-net/issues/9416, https://github.com/Azure/azure-sdk-for-net/issues/13637

### CODE REVIEW

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Amqp/AmqpExceptionHelper.cs [hunk 1]
```diff
@@ -263,5 +263,18 @@ public static Exception GetInnerException(this AmqpObject amqpObject)
 
             return innerException == null ? null : GetClientException(innerException, null, null, connectionError);
         }
+
+        public static bool TryTranslateToRetriableException(Exception exception, out ServiceBusException retriableException)
```
Comment (DorothySun216) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/Amqp/AmqpExceptionHelper.cs:267]: I realize we don't have tests for this class and not sure if I should add a test for this method. I took this class from Serkant's PR fix.

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Core/MessageSender.cs [hunk 2]
```diff
@@ -573,7 +573,14 @@ async Task OnSendAsync(IList<Message> messageList)
...
                 catch (Exception exception)
                 {
-                    throw AmqpExceptionHelper.GetClientException(exception, amqpLink?.GetTrackingId(), null, amqpLink?.Session.IsClosing() ?? false);
+                    if (AmqpExceptionHelper.TryTranslateToRetriableException(exception, out var retriableEx))
+                    {
+                        throw retriableEx;
+                    }
+                    else
+                    {
+                        throw AmqpExceptionHelper.GetClientException(exception, amqpLink?.GetTrackingId(), null, amqpLink?.Session.IsClosing() ?? false);
```
Comment (DorothySun216) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/Core/MessageSender.cs:582]: Neeraj once made a fix for this Github issue but not sure why it didn't resolve it and similar cases kept coming up: https://github.com/Azure/azure-sdk-for-net/pull/6940/files/1b6a6da7e95ecb0868691ae94948acaa27851b03
We have made sure their SDK had Neeraj's fix. Add the retriable exception to every place where this error might occur to attempt to resolve this incident.

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/MessageSession.cs [hunk 3]
```diff
@@ -105,7 +105,14 @@ protected async Task<byte[]> OnGetStateAsync()
             }
             catch (Exception exception)
             {
-                throw AmqpExceptionHelper.GetClientException(exception);
+                if (AmqpExceptionHelper.TryTranslateToRetriableException(exception, out var retriableEx))
```
Comment (serkantkaraca) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/MessageSession.cs:108]: >TryTranslateToRetriableException [](start = 40, length = 32)

Why not move this translation into AmqpExceptionHelper.GetClientException()?  There is similar conversions done in that function.

Comment (DorothySun216) [reply sdk/servicebus/Microsoft.Azure.ServiceBus/src/MessageSession.cs]: Got it, this is a very good catch. Thanks!

**Review by serkantkaraca — APPROVED** [review]
:shipit:

**Review by yvgopal — APPROVED** [review]
:shipit:

**Comment by ramya-rao-a** [comment]
@JoshLove-msft Can you check if a similar change would be needed for Azure.Messaging.ServiceBus?

**Comment by JoshLove-msft** [comment]
I believe this is being handled here - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs#L1109

**Pull request merged**